x-pack/plugins/security_solution/public/common/components/exceptions/helpers.tsx

```diff
@@ -786,8 +786,8 @@ export const defaultEndpointExceptionItems = (
   ruleName: string,
   alertEcsData: Flattened<Ecs>
 ): ExceptionsBuilderExceptionItem[] => {
-  const { event: alertEvent } = alertEcsData;
-  const eventCode = alertEvent?.code ?? '';
+  // TODO: make a ticket to fix this with common type
+  const eventCode = (alertEcsData as Flattened<Ecs> & { 'event.code': string })['event.code'];
```

Comment:
I don't think you need to cast here. Could you just update the prop type to be something like:

```suggestion
  alertEcsData: Flattened<Ecs> & { 'event.code': string }
```

Also, is this value guaranteed? If not, could we add a test (sorry if it's there and I missed it) for what happens when it is not?

Reply:
The field is guaranteed on endpoint alerts (dot notation or nested), a little further down the file we have [a default](https://github.com/elastic/kibana/pull/121720/files/1625d98a08e9a3c4cf85cc67f5127e1f1e0913bd#diff-6808c52ab81432d126e55b4a4d4ff277c9fb83276e3f9170ca8f0ae1ce922efcR830) though, and there's a test already for it